silkworm/rpc/http/connection.hpp

```diff
@@ -70,17 +70,22 @@ class Connection : public StreamWriter {
     AuthorizationResult is_request_authorized(const boost::beast::http::request<boost::beast::http::string_body>& req);
 
     Task<void> handle_request(const boost::beast::http::request<boost::beast::http::string_body>& req);
+    Task<void> handle_actual_request(const boost::beast::http::request<boost::beast::http::string_body>& req);
+    Task<void> handle_preflight(const boost::beast::http::request<boost::beast::http::string_body>& req);
+
+    bool is_origin_allowed(const std::vector<std::string>& allowed_origins, std::string origin);
+    bool is_method_allowed(boost::beast::http::verb method);
 
     Task<void> do_upgrade(const boost::beast::http::request<boost::beast::http::string_body>& req);
 
     template <class Body>
-    void set_cors(boost::beast::http::response<Body>& res);
+    void set_cors(const boost::beast::http::request<boost::beast::http::string_body>& req, boost::beast::http::response<Body>& res);
```

Comment:
The `req` parameter seems redundant here, given that you already store the last request and you could split the `set_cors` code moving the parts relevant to pre-flight in `handle_preflight`